BismarkUnchained/scenes/UI/Upgrades.gd

```diff
@@ -0,0 +1,89 @@
+extends Control
+
+var line
+var to_save = false
+
+# Passive upgrades
+var upgrades = 0
+var maxhealth = 100
+var regen = 0
+var lifesteal = 0
+var rof = 0.75
+
+# Upgrade values
+const rof_upgrade = 0.9
+const maxhealth_upgrade = 25
+const regen_upgrade = 2
+const lifesteal_upgrade = 2
+
+func _ready():
+	var save_file = File.new()
+	if save_file.file_exists("res://bismarkunchained.sav"):
+		save_file.open("res://bismarkunchained.sav", File.READ)
+		line = parse_json(save_file.get_line())
+		upgrades = line.upgrades
+		maxhealth = line.maxhealth
+		regen = line.regen
+		lifesteal = line.lifesteal
+		rof = line.rof
+		save_file.close()
```

Comment:
is there a brittle error checker we can use (i.e. try to parse JSON)? I can see file read errors happening, and in that case we might as well just wipe the file and reset